bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/TcsControllerEpics.scala

```diff
@@ -111,19 +113,26 @@ object TcsControllerEpics extends TcsController {
       ca <- g.nodcchopa.map(decodeNodChopOption)
       cb <- g.nodcchopb.map(decodeNodChopOption)
       cc <- g.nodcchopc.map(decodeNodChopOption)
-    } yield {
-      if (List(aa, ab, ac, ba, bb, bc, ca, cb, cc).exists(x => x)) {
-        if (List(aa, bb, cc).forall(x => x) && !List(ab, ac, ba, bc, ca, cb).exists(x => x))
-          NodChopTrackingConfig.Normal
-        else NodChopTrackingConfig.Special(List((aa, NodChop(Beam.A, Beam.A)), (ab, NodChop(Beam.A, Beam.B)),
-          (ac, NodChop(Beam.A, Beam.C)), (ba, NodChop(Beam.B, Beam.A)), (bb, NodChop(Beam.B, Beam.B)),
-          (bc, NodChop(Beam.B, Beam.C)), (ca, NodChop(Beam.C, Beam.A)), (cb, NodChop(Beam.C, Beam.B)),
-          (cc, NodChop(Beam.C, Beam.C))
-        ).foldLeft(Set[NodChop]())((s: Set[NodChop], v: (Boolean, NodChop)) => if (v._1) s + v._2 else s))
-      }
-      else NodChopTrackingConfig.None
-    }
-  }
+
+      // This last production is slightly tricky. 
+      o  <- if (List(aa, ab, ac, ba, bb, bc, ca, cb, cc).exists(identity)) {
```

Comment:
Actually I'm just going to replace these with `(_ == true)` etc. for now; I don't think the fold makes it any clearer.